src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java

```diff
@@ -451,7 +452,8 @@ private BackOffPolicy getBackoffPolicy(Backoff backoff, boolean stateless) {
 		boolean isRandom = false;
 		String randomExpression = (String) attrs.get("randomExpression");
 		Expression parsedRandomExp = null;
-		if (multiplier > 0) {
```

Comment:
I think the problem is exactly with this `if`.
See docs on those annotation attributes:
```
	/**
	 * Evaluates to a value used as a multiplier for generating the next delay for
	 * backoff; ; any value less than or equal to 1.0 is treated as 1.0, meaning a fixed
	 * delay. Overrides {@link #multiplier()}. Use {@code #{...}} for one-time evaluation
	 * during initialization, omit the delimiters for evaluation at runtime.
	 * @return a multiplier expression to use to calculate the next backoff delay (default
	 * 0 = ignored)
	 * @since 1.2
	 */
	String multiplierExpression() default "";
```

So, if you don't use SpEL template, then it is expected to be evaluated at runtime for each call, therefore that `multiplier` is never going to be `> 0` and we just don't determine anything for `random`.

Reply:
Hi @artembilan , Thank you so much for your replies. 

The multiplierExpression was allowing values between 0 to 1 and was not treating it as 1. So if you provide a expression 0.5 then you will be able to see the retry time getting decreased. I added fix for that as well. 
You can see example of that issue being replicated here: [spring-retry-issue](https://github.com/aftabshk/spring-retry-issue)

Do you expect anything else on this?